aws/codedeploy_workflow/deploy/policy_codepipeline_role.tf

```diff
@@ -0,0 +1,64 @@
+data "aws_iam_policy_document" "code_pipeline_role_policy" {
+  count = var.enable_codepipeline ? 1 : 0
+  statement {
+    actions = [
+      "s3:GetBucketVersioning"
+    ]
+    resources = [
+      data.aws_s3_bucket.deployment.arn,
+      "${module.pipeline_bucket[0].s3_bucket_arn}/*"
+    ]
+  }
+
+  statement {
+    actions = [
+      "s3:GetObject",
+      "s3:GetObjectVersion"
+    ]
+    resources = [
+      "${data.aws_s3_bucket.deployment.arn}/*",
+      "${module.pipeline_bucket[0].s3_bucket_arn}/*"
+    ]
+  }
+
+  statement {
+    actions = [
+      "s3:PutObject"
+    ]
+    resources = [
+      "${module.pipeline_bucket[0].s3_bucket_arn}/*"
+    ]
+  }
+
+  statement {
+    effect = "Allow"
+
+    actions = [
+      "kms:Encrypt",
+      "kms:Decrypt",
+      "kms:ReEncrypt*",
+      "kms:GenerateDataKey*",
+      "kms:DescribeKey"
+    ]
+
+    resources = [var.key_arn]
+  }
+
+  statement {
+    actions = [
+      "codedeploy:CreateDeployment",
+      "codedeploy:GetApplicationRevision",
+      "codedeploy:GetDeployment",
+      "codedeploy:GetDeploymentConfig",
+      "codedeploy:RegisterApplicationRevision"
+    ]
+    resources = ["*"]
```

Comment:
Can this be narrowed down at all?

Reply:
probably

Reply:
I've added the following:

```
    resources = [
      aws_codedeploy_app.deploy_app.arn,
      aws_codedeploy_deployment_group.deployment_group.arn
    ]
```

The reason is because I've updated the feature switches to force the creation of the codedeploy app since it's essentially the entire purpose of the module. We were getting too many logic contradictions otherwise. The codepipeline portion of it still remains optional.